File: JavaVsKotlin_1/app/src/main/java/com/example/javavskotlin_1/MainActivity.java
package com.example.javavskotlin_1;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import com.example.kotlinvsjava_1.Car;
import com.example.kotlinvsjava_1.KotlinClass;

public class MainActivity extends AppCompatActivity {
    TextView textView1;
    Button button1;
    Button button2;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textView1 = findViewById(R.id.textView1);
        button1 = findViewById(R.id.buttonJava);
        button2 = findViewById(R.id.buttonKotlin);

        textView1.setText("Text View yang baru tidak perlu untuk di cash pda Find View by ID.");
        button1.setOnClickListener(e ->{
            textView1.setText("Hello From Java Biasa");

        });
        button2.setOnClickListener(e -> {
            Car car1 = new Car("Mas Bagus Winarno");
            Car car2 = new Car(1, null);
//            Car car3 = new Car(null, 2); //Tidak bisa karena tanda "?" berarti boleh null
            KotlinClass kc = new KotlinClass();

            Toast.makeText(this, kc.setTextFromKotlin_WitMesageReturn(textView1) + " dan ", Toast.LENGTH_LONG).show();
        });

    }


}

File: KotlinVsJava_1/app/src/main/java/com/example/kotlinvsjava_1/MainActivity.kt
package com.example.kotlinvsjava_1

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
//    var textView1: TextView? = null
//    var button1: Button? = null
//    var button2: Button? = null
    val textView1 = findViewById(R.id.textView1) as TextView
    val button1 = findViewById(R.id.buttonJava) as Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

//        textView1 = findViewById(R.id.textView1)
//        button1 = findViewById(R.id.buttonJava)
//        button2 = findViewById(R.id.buttonKotlin)

        textView1.text = getString(R.string.app_name)
        button1?.setOnClickListener {
            textView1.setText("Hello Java")
            textView1.text = "Hello Java (Kotlin)"
        }


//        var nilai:String = textView1.text.toString()

    }

    fun test(){
        textView1.text = "Hello World"
    }


}